### PR TITLE
BUGFIX: Stabilize doctrine migrations

### PR DESCRIPTION
**Pull request by bwaidelich**
Related: #3118

### CODE REVIEW

**Comment by jonnitto** [comment]
How can I test this one?

**Comment by sorenmalling** [comment]
@bwaidelich Is similar required for pgsql?

**Review by markusguenther — APPROVED** [review]

**Comment by markusguenther** [comment]
> @bwaidelich Is similar required for pgsql?

For Postgres, we have own migrations

**Review by kdambekalns — COMMENTED** [review]
TBH: I am too lazy to try this out, so let me just ask: Not all migrations have `up` _and_ `down` adjusted, is that correct? Does it work in both directions?

**Comment by bwaidelich** [comment]
> TBH: I am too lazy to try this out

Haha, thanks for the honest reply :D

I'll test down migration and take care of postgres and steps to reproduce asap

**Review by fcool — APPROVED** [review]
At least the up migration part for MySQL works as expected, if this patch is applied.

Tested it in several cases (empty database, already existing flow project, where neos is added right now) and everytime everything was the way it should be. :)

File: Neos.Media/Migrations/Mysql/Version20141118172322.php [hunk 1]
```diff
@@ -24,7 +24,10 @@ public function up(Schema $schema): void
...
         $this->addSql("ALTER TABLE typo3_media_domain_model_imagevariant ADD CONSTRAINT FK_758EDEBD47A46B0A FOREIGN KEY (persistence_object_identifier) REFERENCES typo3_media_domain_model_asset (persistence_object_identifier) ON DELETE CASCADE");
         $this->addSql("ALTER TABLE typo3_media_domain_model_thumbnail ADD CONSTRAINT FK_B7CE141455FF4171 FOREIGN KEY (originalasset) REFERENCES typo3_media_domain_model_asset (persistence_object_identifier)");
-        $this->addSql("ALTER TABLE typo3_media_domain_model_thumbnail ADD CONSTRAINT FK_B7CE1414BC91F416 FOREIGN KEY (resource) REFERENCES typo3_flow_resource_resource (persistence_object_identifier)");
+        $tableNames = $this->sm->listTableNames();
+        if (in_array('typo3_flow_resource_resource', $tableNames, true) && in_array('typo3_typo3_domain_model_media_image', $tableNames, true)) {
+            $this->addSql("ALTER TABLE typo3_typo3_domain_model_media_image ADD CONSTRAINT typo3_typo3_domain_model_media_image_ibfk_1 FOREIGN KEY (flow3_resource_resource) REFERENCES typo3_flow_resource_resource(flow3_persistence_identifier)");
+        }
```
Comment (kdambekalns) [inline Neos.Media/Migrations/Mysql/Version20141118172322.php:30]: Other similar checks have an `elseif` part, is that not needed here?

Comment (fcool) [reply Neos.Media/Migrations/Mysql/Version20141118172322.php:30]: At least I can confirm that the difference to the "current" schema after a `doctrine:migrate` has been the same in all cases.

Are we aware that even a fully migrated Neos CMS 7.3 creates new Migrations for two tables? (`neos_contentrepository_domain_model_nodedata` and `neos_media_domain_model_tag`)

Comment (fcool) [reply Neos.Media/Migrations/Mysql/Version20141118172322.php:30]: (but that's not the business of this PR ;) )

**Comment by bwaidelich** [comment]
This targets an outdated branch, we can pick this one up again if issues pop up

**Comment by fcool** [comment]
How about just retargeting it? The issue is valid even for 8.3 - which is no wonder, as the migrations usally do never change ;)

**Comment by kdambekalns** [comment]
I think _something_ needs to be done, so I'll reopen and assign to myself. I'll check the needed things for 7.3 and up the coming two weeks If I don't, please nag me.

**Comment by mhsdesign** [comment]
@kdambekalns did your fix over here https://github.com/neos/neos-development-collection/pull/4236 makes this obsolete?

**Comment by bwaidelich** [comment]
@kdambekalns *ping*

I'll close this one for now to remove it from my task list – feel free to re-open if it's still valuable

**Pull request closed without merging**